server/lib/fetch.py

```diff
@@ -310,10 +312,25 @@ def triples(nodes, out=True):
   }
 
   """
-  resp = dc.triples(nodes, 'out' if out else 'in')
+  resp = dc.v2node(nodes, '->*' if out else '<-*')
   result = {}
   for node, arcs in resp['data'].items():
     result[node] = {}
     for prop, val in arcs.get('arcs', {}).items():
       result[node][prop] = val.get('nodes', [])
-  return result
\ No newline at end of file
+  return result
+
+
+def descendent_places(nodes, descendent_type):
+  return property_values(nodes,
+                         'containedInPlace+',
```

Comment:
wonder if it would be better to handle the "+" in the property_values function instead so that the caller just has to pass the proper properties and constraints without needing to know how to build the query string? Same with raw_property_values

Reply:
+ is part of the composite property. If making it an argument of property_values(), the caller would still need to set it.

This is the reason to have this wrapper function descendent_places(), and all actual callers should just call this function, and would never need to know about "+"

Reply:
by handle it in property_values, I meant property_values would still take the same arguments but add a "+" if there are constraints (because from what I understand, we add a "+" if there are constraints?)

Reply:
In the V2 syntax, "+" is like regex, meaning 1 or more chained prop, so in this case it is:
<-conainedInPlace<-conainedInPlace<-...<-containedInPlace.

The constraints {p:v} is independent from that.. meaning any node in the chain that meets the pv constraints.

Reply:
ohh I see, ok, that makes sense!